server/src/main/java/org/elasticsearch/features/NodeFeature.java

```diff
@@ -15,10 +15,17 @@
  * A feature published by a node.
  *
  * @param id        The feature id. Must be unique in the node.
+ * @param assumedAfterNextCompatibilityBoundary
+ *              {@code true} if this feature is removed at the next compatibility boundary (ie next major version),
+ *              and so should be assumed to be true for all nodes after that boundary.
  */
-public record NodeFeature(String id) {
+public record NodeFeature(String id, boolean assumedAfterNextCompatibilityBoundary) {
```

Comment:
Naming might need a bit of work

Reply:
I struggle coming up with something better... `assumePresenceAfterNextCompatibilityBoundary`, but that's even harder to read.... `assumedOnNextMajor` maybe?

Reply:
Yeah, exactly, it was `assumedOnNextMajor`, but it's always true for serverless, so...